app/assets/javascripts/routers/data_portal.js

```diff
@@ -28,6 +29,37 @@
         iso: iso,
         year: +year
       });
+    },
+
+    report: function (params) {
+      // Don't forget to stop the router on each route
+      // otherwise you'll break the browser's back button
+      Backbone.history.stop();
+      var indicators = [],
+        serializedIndicators;
+
+      if (params) {
+        var indicatorsb64 = params.split('=')[1] ? params.split('=')[1] : null;
+        if (indicatorsb64) {
+          // decodes b64 string to object
+          indicators = JSON.parse(App.Helper.URL.decode(indicatorsb64).indicators);
```

Comment:
You shouldn't need to parse here because `decode` already do it.

The issue is in `ReportFixedBarView.js.erb` because you encode a JS object with the list of _stringified_ indicators. Instead pass the real list of indicators:
```js
_encodeIndicators: function () {
  return App.Helper.URL.encode({ indicators: JSON.parse(localStorage.getItem('indicators')) });
},
```